sr/comp/cli/show_match_scores.py

```diff
@@ -0,0 +1,167 @@
+
+from typing import Dict, List, Union, NamedTuple
+from sr.comp.types import TLA, GamePoints, ArenaName, MatchNumber
+from sr.comp.scores import LeaguePosition
+
+
+class MatchCorner(NamedTuple):
+    tla: TLA
+    ranking: Union[LeaguePosition, str]  # to allow "???" to be used for unknown scores
+    game: Union[GamePoints, str]
+    league: Union[int, str]
+
+
+class MatchResult(NamedTuple):
+    num: MatchNumber
+    arena: ArenaName
+    display_name: str
+    corners: Dict[int, MatchCorner]
+
+
+def collect_match_info(comp, match) -> MatchResult:
+    from sr.comp.match_period import MatchType
+    from sr.comp.scores import degroup
+
+    if match.type == MatchType.knockout:
+        score_data = comp.scores.knockout
+    elif match.type == MatchType.tiebreaker:
+        score_data = comp.scores.tiebreaker
+    elif match.type == MatchType.league:
+        score_data = comp.scores.league
+
+    match_id = (match.arena, match.num)
+
+    if match_id in score_data.game_points:
+        league_points = score_data.ranked_points[match_id]
+        game_points = score_data.game_points[match_id]
+        ranking = degroup(score_data.game_positions[match_id])
+    else:
+        league_points = {}
+        game_points = {}
+        ranking = {}
+        for team in match.teams:
+            league_points[team] = "???"
+            game_points[team] = "???"
+            ranking[team] = "???"
+
+    corner_data: Dict[int, MatchCorner] = {}
+
+    for corner, team in enumerate(match.teams):
+        match_id = (match.arena, match.num)
+
+        if team:  # corner occupied
+            corner_data[corner] = MatchCorner(
+                tla=team,
+                ranking=ranking[team],
+                game=game_points[team],
+                league=league_points[team],
+            )
+        else:
+            corner_data[corner] = MatchCorner(
+                tla='',
+                ranking='',
+                game='',
+                league='',
+            )
+
+    return MatchResult(
+        num=match.num,
+        arena=match.arena,
+        display_name=match.display_name,
+        corners=corner_data,
+    )
+
+
+def print_col(text):
+        print(text, end='|')
+
+
+def print_heading(num_corners, name_width, arena_name_width):
+    print_col("".center(name_width + 1 + arena_name_width))
+    for idx in range(num_corners):
+        print_col(f"Zone {idx}".center(22))
+    print()
+
+    print_col("Display Name".center(name_width))
+    print_col("Arena".center(arena_name_width))
+    for idx in range(num_corners):
+        print_col("TLA".center(5))
+        print_col("Rank")
+        print_col("Game")
+        print_col("League")
+    print()
+
+
+def print_match(match: MatchResult, name_width, arena_name_width):
+    print_col(match.display_name.center(name_width))
+    print_col(match.arena.center(arena_name_width))
+
+    for corner in match.corners.values():
```

Comment:
Feels like this is going to get pretty wide even just for a four zone match. Is this the best layout for a CLI tool?

Reply:
With the dummy comp the table was 120 wide, this increased to 146 when I moved to using `tabulate` so I've made it wrap each match at 2 zones in 52931ce15f79b651ad4ad36c8261ce46b1422713 which gives a width of 100.

At this point the only part that effects the width of the table is the match and arena names. The next version of `tabulate` looks to be adding and option to force wrap certain columns at a set width.

Alternatively if we can come up with a way to merge the game and league columns they are limited by the length of the heading. In fact omitting all the headers reduces the width to 78.

```
+-----------------------+------+------+------+------+--------+------+------+------+------+--------+
|         Match         | Zone | TLA  | Rank | Game | League | Zone | TLA  | Rank | Game | League |
+-----------------------+------+------+------+------+--------+------+------+------+------+--------+
|     Match 0 in A      |    0 |      |      |      |        |    1 | CLY  |    1 |    9 |      8 |
|                       |    2 | TTN  |    2 |    6 |      6 |    3 |      |      |      |        |
|     Match 0 in B      |    0 | GRS  |    1 |    5 |      8 |    1 | QMC  |    2 |    3 |      6 |
|                       |    2 |      |      |      |        |    3 |      |      |      |        |
|     Match 1 in A      |    0 | WYC  |    4 |    3 |      2 |    1 | QMS  |    2 |    7 |      6 |
|                       |    2 | LSS  |    1 |   10 |      8 |    3 | EMM  |    3 |    5 |      4 |
|     Match 1 in B      |    0 | BPV  |    2 |    6 |      6 |    1 | BDF  |    4 |    2 |      2 |
|                       |    2 | NHS  |    1 |    7 |      8 |    3 | MEA  |    3 |    4 |      4 |
| Quarter 1 (#123) in A |    0 | ???  |   ?? |   ?? |     ?? |    1 | ???  |   ?? |   ?? |     ?? |
|                       |    2 | ???  |   ?? |   ?? |     ?? |    3 | ???  |   ?? |   ?? |     ?? |
| Quarter 2 (#124) in A |    0 | ???  |   ?? |   ?? |     ?? |    1 | ???  |   ?? |   ?? |     ?? |
|                       |    2 | ???  |   ?? |   ?? |     ?? |    3 | ???  |   ?? |   ?? |     ?? |
| Quarter 3 (#125) in A |    0 | ???  |   ?? |   ?? |     ?? |    1 | ???  |   ?? |   ?? |     ?? |
|                       |    2 | ???  |   ?? |   ?? |     ?? |    3 | ???  |   ?? |   ?? |     ?? |
| Quarter 4 (#126) in A |    0 | ???  |   ?? |   ?? |     ?? |    1 | ???  |   ?? |   ?? |     ?? |
|                       |    2 | ???  |   ?? |   ?? |     ?? |    3 | ???  |   ?? |   ?? |     ?? |
```

Reply:
Hrm. I definitely think we shouldn't mix wrapping with putting things next to each other -- so if we're having a line per zone we should always have that. Both in the sense that the output shouldn't mix the two styles, but also that we shouldn't try to guess which style to use. I can maybe see an argument for letting the user choose, however in that case I think we should ignore the width that that ends up being (since that's then the user's problem -- they got what they asked for).

I think for a first pass here we should go with a simple approach which would always work. This probably means roughly the style you've suggested here, but only one zone wide. (I think this would also simplify the code a bit?)

I think I'd also push the arena name back into its own column? Not sure how that would look in the two-arena case though.